File: math/root/digital.c
int mathematical(int n)
{
    return n %9 ? n %9 : n ? 9 : n;
}
int recursive(int n)
{
    if(n <10) return n;

    int sum = 0;

    while(n)
    {
        sum += n %10;
        n /= 10;
    }
    return recursive(sum);
}
int iterative(int n)
{
    int current;
    int sum;

    while(9 <n)
    {
        current = n;
        sum = 0;

        while(current)
        {
            sum += current %10;
            current /= 10;
        }
        n = sum;
    }
    return n;
}

File: math/root/digital.js
function mathematical(n) {
    return (r = n %9) ? r : n ? 9 : n
}
function recursive(n) {
    return n <10 ? n : recursive(Array.prototype.reduce.call(n.toString(),(A,V) => Number(V) +A,0))
}
function iterative(n) {
    var sum;

    while(9 <n) {
        sum = 0;

        while(n) {

            sum += n %10;
            n = Math.floor(n /10);
        }
        n = sum;
    }
    return n;
}